include/benchmark/benchmark.h

```diff
@@ -385,27 +385,27 @@ class Counter {
     kDefaults = 0,
     // Mark the counter as a rate. It will be presented divided
     // by the duration of the benchmark.
-    kIsRate = 1U << 0U,
+    kIsRate = 1 << 0,
     // Mark the counter as a thread-average quantity. It will be
     // presented divided by the number of threads.
-    kAvgThreads = 1U << 1U,
+    kAvgThreads = 1 << 1,
     // Mark the counter as a thread-average rate. See above.
     kAvgThreadsRate = kIsRate | kAvgThreads,
     // Mark the counter as a constant value, valid/same for *every* iteration.
     // When reporting, it will be *multiplied* by the iteration count.
-    kIsIterationInvariant = 1U << 2U,
+    kIsIterationInvariant = 1 << 2,
     // Mark the counter as a constant rate.
     // When reporting, it will be *multiplied* by the iteration count
     // and then divided by the duration of the benchmark.
     kIsIterationInvariantRate = kIsRate | kIsIterationInvariant,
     // Mark the counter as a iteration-average quantity.
     // It will be presented divided by the number of iterations.
-    kAvgIterations = 1U << 3U,
+    kAvgIterations = 1 << 3,
     // Mark the counter as a iteration-average rate. See above.
     kAvgIterationsRate = kIsRate | kAvgIterations,
 
     // In the end, invert the result. This is always done last!
-    kInvert = 1U << 31U
+    kInvert = 1 << 31
```

Comment:
Isn't `1 << 31` UB on platforms with 32-bit signed int? Instead, would it make sense to write `enum Flags : uint32_t` and bring back `1U << 31U` here?

Reply:
I believe you are correct in stating that `1 << 31` is UB. I must have mistaken an enum's ability to account for this when it [fits all underlying values](https://timsong-cpp.github.io/cppwp/n4659/dcl.enum#7).

I also tried a similar solution as you are mentioning when I originally made a patch (see https://github.com/alisenai/benchmark/commit/e75182f7edfd702020f668f72ee142d29c6efad1), though I do not recall why I decided against it (?).

Reply:
Oh, right, as the `README.md` states:
> The library can be used with C++03 (...)

Since enum type specifiers were only added in c++11 (correct me if I'm wrong), `enum Flags : uint32_t` would not compile when including this header in a C++03 file/project.

Though, looking through the rest of the file, maybe a similar solution can be implemented?https://github.com/google/benchmark/blob/60b16f11a30146ac825b7d99be0b9887c24b254a/include/benchmark/benchmark.h#L577-L581